src/main/java/nextstep/security/filter/BasicAuthFilter.java

```diff
@@ -0,0 +1,87 @@
+package nextstep.security.filter;
+
+import jakarta.servlet.Filter;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.ServletRequest;
+import jakarta.servlet.ServletResponse;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import nextstep.security.AuthenticationException;
+import nextstep.security.UserDetailService;
+import nextstep.security.UserDetails;
+import nextstep.security.util.Base64Convertor;
+
+import java.util.List;
+
+import static org.springframework.http.HttpHeaders.AUTHORIZATION;
+
+public class BasicAuthFilter implements Filter {
+
+    private static final List<String> TARGET_URL = List.of("/members");
+
+    private final UserDetailService userDetailsService;
+
+    public BasicAuthFilter(UserDetailService userDetailsService) {
+        this.userDetailsService = userDetailsService;
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain) throws ServletException {
+        boolean isHttpServlet = servletRequest instanceof HttpServletRequest && servletResponse instanceof HttpServletResponse;
+        if (isHttpServlet) {
+            HttpServletRequest request = (HttpServletRequest) servletRequest;
+            HttpServletResponse response = (HttpServletResponse) servletResponse;
+
+            try {
+                boolean notTarget = isNotTarget(request);
+                if (notTarget) {
+                    filterChain.doFilter(request, response);
+                    return;
+                }
+
+                checkAuthentication(request);
+                filterChain.doFilter(request, response);
+            } catch (Exception e) {
+                response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+            }
+            return;
+        }
+        throw new ServletException("BasicAuthFilter only supports HTTP requests");
```

Comment:
코드의 depth가 길어지면 코드의 가독성이 떨어지게 되는데요.
그런 관점에서 early return하는 코드를 작성한다면 가독성이 올라갈 것 같습니다!

```suggestion
        if (!isHttpServlet) {
                throw new ServletException("BasicAuthFilter only supports HTTP requests");
        }

        HttpServletRequest request = (HttpServletRequest) servletRequest;
        HttpServletResponse response = (HttpServletResponse) servletResponse;

        try {
            boolean notTarget = isNotTarget(request);
            if (notTarget) {
                filterChain.doFilter(request, response);
                return;
            }

            checkAuthentication(request);
            filterChain.doFilter(request, response);
        } catch (Exception e) {
            response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
        }

```

Reply:
피드백 반영하겠습니다! 감사합니다 :)
이번 미션에서는 기본적으로 security의 동작 원리 중심으로 보고있어 신경을 쓰지 못했지만 가독성도 고려해서 작성해보겠습니다! :)